interfaces/acados_template/acados_template/utils.py

```diff
@@ -165,14 +165,25 @@ def make_model_consistent(model):
 
     return model
 
-def get_lib_ext():
-    lib_ext = '.so'
+def get_shared_lib_ext():
     if sys.platform == 'darwin':
-        lib_ext = '.dylib'
+        return '.dylib'
     elif os.name == 'nt':
-        lib_ext = ''
+        return '.dll'
+    else:
+        return '.so'
+
+def get_shared_lib_dir():
+    if os.name == 'nt':
+        return 'bin'
```

Comment:
I am a bit confused by this. Can you explain?
We should also test Windows on Matlab before merging this.
As far as I remember, it expects the libs to be in `lib`.

Reply:
Windows, and particularly MSVC, behaves a bit different than Linux when it comes to shared libraries. Windows also uses so called import libraries, containing stubs to the exported functions of the shared library. 

- Import libraries (`.lib` on windows - confusingly just like static libraries are `.lib`) go to the `lib\` folder and are needed at link time only.
- Shared libraries (`.dll` on windows) go to the `bin\` folder and are needed at runtime only.

Different compilers have different ways of dealing with them:
- MSVC cannot link directly to a `.dll` shared library. At link time it really needs the `.lib` import library.
- MinGW doesn't really need import libraries, as it can directly link to a `.dll`, but - I guess for uniformity - it can also handle import libraries (which it would typically give the extension `.dll.a`, but I renamed that suffix in CMake to `.lib`).

So:
- At link time, the `.lib` import libraries in the `lib\` folder are required, also for e.g. `mex`-ing with Matlab.
- At run time, the `.dll` shared libraries in the `bin\` folder are required. The Python DLL loader (`ctypes.WinDLL`) is reading the shared libraries at run time, so needs to get them from the `bin\` folder.